httpclient5/src/main/java/org/apache/hc/client5/http/impl/async/InternalAbstractHttpAsyncClient.java

```diff
@@ -104,6 +110,7 @@
         this.credentialsProvider = credentialsProvider;
         this.defaultConfig = defaultConfig;
         this.closeables = closeables != null ? new ConcurrentLinkedQueue<>(closeables) : null;
+        this.scheduledExecutorService = Executors.newSingleThreadScheduledExecutor(new DefaultThreadFactory("Scheduled-executor"));
```

Comment:
Maybe the factory should be distinguishable between client instances in one VM...?

Reply:
@michael-o I will make the factory static.